crates/weaver_resolver/src/body.rs

```diff
@@ -0,0 +1,88 @@
+
+// SPDX-License-Identifier: Apache-2.0
+
+//! Functions to resolve a semantic convention body.
+
+use weaver_resolved_schema::body::{Body, BodyField};
+use weaver_semconv::body::{BodyFieldSpec, BodySpec};
+
+use crate::attribute::AttributeCatalog;
+
+/// Resolve a body specification into a resolved body.
+pub fn resolve_body(
+    body: &BodySpec,
+    attr_catalog: &mut AttributeCatalog
+) -> Option<Body> {
+    match body {
+        BodySpec::Fields { fields } => {
+            let fields = fields.iter().map(|field| resolve_body_field(field, attr_catalog)).collect();
+            Some(Body {
+                fields: Some(fields),
+                value: None,
+            })
+        },
+        BodySpec::Value { value: _ } => {
+            // Add as a placeholder for now of where to resolve the value.
+            todo!("Implement value type for body.")
+        }
+    }
+}
+
+/// Resolve a body field specification into a resolved body field.
+pub fn resolve_body_field(
+    field: &BodyFieldSpec,
+    attr_catalog: &mut AttributeCatalog
+) -> BodyField {
+    match field {
+        BodyFieldSpec::Ref {
+            r#ref,
+            alias,
+            brief,
+            examples,
+            requirement_level,
+            note,
+            stability,
+            deprecated,
+        } => {
+            let attr_ref = attr_catalog.get_attribute_ref(r#ref);
```

Comment:
I'm not entirely sure I understand the difference between an attribute and a field, but isn't it possible to have a field with the same ID as an attribute? If so, it would probably be preferable to implement a FieldCatalog similar to the AttributeCatalog.